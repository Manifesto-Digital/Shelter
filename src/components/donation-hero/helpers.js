export const formatDonationContent = content => {
  const createOptionObject = (amount, description, image) => ({
    amount,
    description,
    image,
  });

  // single
  const { singleAmount1, singleDescription1, singleImage1 } = content;
  const singleOption1 = createOptionObject(
    singleAmount1,
    singleDescription1,
    singleImage1
  );
  const { singleAmount2, singleDescription2, singleImage2 } = content;
  const singleOption2 = createOptionObject(
    singleAmount2,
    singleDescription2,
    singleImage2
  );
  const { singleAmount3, singleDescription3, singleImage3 } = content;
  const singleOption3 = createOptionObject(
    singleAmount3,
    singleDescription3,
    singleImage3
  );
  const { singleAmount4, singleDescription4, singleImage4 } = content;
  const singleOption4 = createOptionObject(
    singleAmount4,
    singleDescription4,
    singleImage4
  );

  const { singleAmount5, singleDescription5, singleImage5 } = content;
  const singleOption5 = createOptionObject(
    singleAmount5,
    singleDescription5,
    singleImage5
  );

  // monthly
  const { monthlyAmount1, monthlyDescription1, monthlyImage1 } = content;
  const monthlyOption1 = createOptionObject(
    monthlyAmount1,
    monthlyDescription1,
    monthlyImage1
  );
  const { monthlyAmount2, monthlyDescription2, monthlyImage2 } = content;
  const monthlyOption2 = createOptionObject(
    monthlyAmount2,
    monthlyDescription2,
    monthlyImage2
  );
  const { monthlyAmount3, monthlyDescription3, monthlyImage3 } = content;
  const monthlyOption3 = createOptionObject(
    monthlyAmount3,
    monthlyDescription3,
    monthlyImage3
  );
  const { monthlyAmount4, monthlyDescription4, monthlyImage4 } = content;
  const monthlyOption4 = createOptionObject(
    monthlyAmount4,
    monthlyDescription4,
    monthlyImage4
  );
  return {
    single: [
      singleOption1,
      singleOption2,
      singleOption3,
      singleOption4,
      singleOption5,
    ],
    monthly: [monthlyOption1, monthlyOption2, monthlyOption3, monthlyOption4],
  };
};
