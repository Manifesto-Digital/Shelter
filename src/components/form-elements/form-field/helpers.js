const possibleTextInputs = Object.freeze([
  'email',
  'text-field',
  'phone-number',
  'numeric',
  'text-area',
]);

export const isTextField = fieldType => possibleTextInputs.includes(fieldType);

export const getTextInputType = fieldType => {
  if (fieldType === 'phone-number') return 'tel';
  if (fieldType === 'numeric') return 'number';
  if (fieldType === 'email') return fieldType;
  if (fieldType === 'text-area') return 'textarea';
  return 'text';
};
