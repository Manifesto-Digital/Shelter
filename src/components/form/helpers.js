import * as Yup from 'yup';
import { consistentString } from '../../utils/content-formatting';

const isMultipleCheckbox = field =>
  consistentString(field.fieldType) === 'checkboxes' &&
  field.defaultValue &&
  field.valueOptions.length > 1;

export const getInitialValues = (fields, hiddenInitialValues) => {
  const initialValues = hiddenInitialValues;

  // Loop through all fields to build an object to store initial values to pass to formik
  fields.forEach(field => {
    if (!field.fieldType) return;

    if (field.fieldType === 'Address') return; // Temporary

    // Set default value to be the initial value for form
    if (field.internal.type === 'ContentfulTopicFormField') {
      initialValues[field.machineName] = field.defaultValue || '';
    }
    // Multiple checkboxes require an array of values
    if (isMultipleCheckbox(field)) {
      initialValues[field.machineName] = field.defaultValue
        ? [field.defaultValue]
        : [];
    }

    if (field.internal.type === 'ContentfulTopicFormFieldset') {
      getInitialValues(field.formFields);
    }
  });
  return initialValues;
};

export const getValidationSchema = formFields => {
  const validationSchema = {};

  // Loop through all form fields and build a Yup validationSchema object for formik
  formFields.forEach(field => {
    if (!field.fieldType) return;
    const fieldType = consistentString(field.fieldType);
    if (field.fieldType === 'address') return; // Temporary
    const fieldName = field.machineName;

    let validationType;
    // Apply a validation type to each field
    if (isMultipleCheckbox(field)) validationType = Yup.array();
    else if (fieldType === 'phone-number') {
      validationType = Yup.number().typeError('Please provide a valid number');
    } else if (fieldType === 'email') {
      validationType = Yup.string().email('Please provide a valid email');
    } else validationType = Yup.string();

    // Add required validation if specified
    if (field.required) validationType = validationType.required('Required');

    validationSchema[fieldName] = validationType;
  });

  return Yup.object().shape(validationSchema);
};
