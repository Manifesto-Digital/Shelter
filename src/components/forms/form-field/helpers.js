const possibleTextInputs = Object.freeze([
  'email',
  'text-field',
  'phone-number',
  'numeric',
]);

export const isTextField = fieldType => possibleTextInputs.includes(fieldType);

export const getTextInputType = fieldType => {
  if (fieldType === 'phone-number') return 'tel';
  if (fieldType === 'numeric') return 'number';
  if (fieldType === 'email') return fieldType;
  return 'text';
};
