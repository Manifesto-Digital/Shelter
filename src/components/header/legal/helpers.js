/**
 *
 * Loop through the professionals navigation menu to find the current active page
 * and the parent pages so to that the correct menus and active states will be shown
 *
 * @param {Object []} childMenuItems
 * @param {Object} location
 *
 * @returns [menuParentSlug, menuParentSlug, activePageSlug] (example)
 */
export const getActivePages = (childMenuItems, location) => {
  const isActivePage = slug => {
    // Match optional trailing slash
    const string = `${slug}((/w+)+|/?)$`;
    const regex = new RegExp(string, 'g');
    return location.pathname.match(regex);
  };

  let activePagesArray = [];
  let pageFound = false;

  const checkLevel = items => {
    for (let i = 0; i < items.length; i += 1) {
      const item = items[i];
      if (!item.navigationLink) return;

      const itemSlug = item.navigationLink[0].slug;

      if (isActivePage(itemSlug)) {
        // If found push the active page slug
        activePagesArray.push(itemSlug);
        pageFound = true;
        break;
      } else if (item.childNavigationItems) {
        // Recursively go through the child navigation items storing
        // potential active parents item slugs
        activePagesArray.push(itemSlug);
        checkLevel(item.childNavigationItems);
      }
    }
    if (!pageFound) activePagesArray = [];
    return activePagesArray;
  };

  checkLevel(childMenuItems);
  return activePagesArray;
};
