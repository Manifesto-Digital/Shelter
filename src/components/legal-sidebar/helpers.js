export const buildCurrentPageHierarchy = (hierarchy, parentSlug) => {
  const hierarchyDepth = 2;

  // Get the parent of the topmost level to show for use in the heading.
  const headingRef =
    hierarchyDepth > parentSlug.length
      ? parentSlug[parentSlug.length - hierarchyDepth - 1]
      : parentSlug[0];
  const heading = headingRef.label;

  // The top most level is Legal which we do not want to show so remove from parentSlug
  const legalChildrenSlugs = parentSlug.slice(1);

  // Slice the pages parent slug if the current page greater than hierarchy depth
  const topLevelToShowIndex = legalChildrenSlugs.length - hierarchyDepth;
  const parentHierarchyToShow =
    legalChildrenSlugs.length > hierarchyDepth
      ? legalChildrenSlugs.slice(topLevelToShowIndex)
      : legalChildrenSlugs;

  // Check if page exists on current hierarchy level
  function existsOnLevel(pageRef) {
    return parentHierarchyToShow.find(ref => ref.slug === pageRef.slug);
  }

  // Copy to avoid mutating original hierarchy
  const hierarchyCopy = JSON.parse(JSON.stringify(hierarchy));

  // Loop through assign an active state to the current page parents
  function buildActiveHierarchy(hierarchyLevel) {
    const newHierarchy = [];
    function recursiveLoop(level) {
      level.forEach((pageRef, i) => {
        if (existsOnLevel(pageRef)) {
          // Set current item to active
          level[i].active = true;
          // Push the first time this matches, this will also include the children
          if (newHierarchy.length === 0) newHierarchy.push(level);
        }
        if (pageRef.children) {
          recursiveLoop(pageRef.children);
        }
      });
    }
    recursiveLoop(hierarchyLevel);

    return newHierarchy;
  }

  const currentPageHierarchy = buildActiveHierarchy(hierarchyCopy);
  return { currentPageHierarchy: currentPageHierarchy[0], heading };
};
