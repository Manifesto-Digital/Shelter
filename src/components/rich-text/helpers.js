export const fieldsMap = fieldsToMap =>
  Object.keys(fieldsToMap).reduce((accum, key) => {
    const field = fieldsToMap[key];
    if (Object.prototype.hasOwnProperty.call(field, 'en-GB')) {
      accum[key] = flattenKey('en-GB', field); // eslint-disable-line no-use-before-define
    } else if (typeof field === 'object' && field !== null) {
      accum[key] = fieldsMap(field);
    } else {
      accum[key] = field;
    }
    return accum;
  }, {});

const flattenKey = (key, field) => {
  if (typeof field[key] === 'string' || typeof field[key] === 'boolean') {
    return field[key];
  }
  if (Array.isArray(field[key]) && field[key].length === 1) {
    return [fieldsMap(field[key][0].fields)];
  }
  if (typeof field[key] === 'object') {
    if (Object.prototype.hasOwnProperty.call(field[key], 'fields')) {
      return fieldsMap(field[key].fields);
    }

    return field[key];
  }

  return field;
};
