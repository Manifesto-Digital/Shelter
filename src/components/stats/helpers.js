export const statsToArray = content => {
  const createStatObject = (title, subtitle, text) => {
    if (!text) return null;

    return {
      title,
      subtitle,
      text,
    };
  };

  const { stat1Title, stat1Subtitle, stat1Text } = content;
  const stat1 = createStatObject(stat1Title, stat1Subtitle, stat1Text);
  const { stat2Title, stat2Subtitle, stat2Text } = content;
  const stat2 = createStatObject(stat2Title, stat2Subtitle, stat2Text);
  const { stat3Title, stat3Subtitle, stat3Text } = content;
  const stat3 = createStatObject(stat3Title, stat3Subtitle, stat3Text);
  const { stat4Title, stat4Subtitle, stat4Text } = content;
  const stat4 = createStatObject(stat4Title, stat4Subtitle, stat4Text);

  // Remove any nulls as a failsafe
  return [stat1, stat2, stat3, stat4].filter(stat => stat);
};
