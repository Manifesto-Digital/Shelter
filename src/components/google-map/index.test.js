import React from 'react';
import { snapshotComponent } from '../../../__tests__/helpers/index';
import Map from './index';
import { createFactory } from '../../utils/test-factories';

// Default props
export const createLocations = createFactory({
  headerText: 'Topic - Google Map',
  locations: [
    {
      address: 'Bournemouth, UK',
      location: {
        lat: 52.240477,
        lon: -0.902655999999979,
      },
    },
    {
      address: '"Whitefriars House, 50 Fishergate, Norwich, NR3 1SE"',
      location: {
        lat: 52.6344036,
        lon: 1.299995899999999,
      },
    },
  ],
});

it('renders correctly', () => {
  const mockData = createLocations();
  snapshotComponent(<Map data={mockData} />);
});
