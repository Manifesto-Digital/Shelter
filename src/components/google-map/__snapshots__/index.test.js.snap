// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c1 {
  width: 100%;
  height: 400px;
}

.c0 {
  max-width: 1140px;
  margin: 0 auto;
  padding: 0 20px;
}

<div
  className="c0"
>
  <h2>
    Topic - Google Map
  </h2>
  <main
    className="c1"
  >
    <div
      onClick={[Function]}
      onMouseDownCapture={[Function]}
      onMouseMove={[Function]}
      style={
        Object {
          "height": "100%",
          "margin": 0,
          "padding": 0,
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <div
        style={
          Object {
            "height": "100%",
            "left": 0,
            "margin": 0,
            "padding": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      />
      <div
        style={
          Object {
            "height": "50%",
            "left": "50%",
            "margin": 0,
            "padding": 0,
            "position": "absolute",
            "top": "50%",
            "width": "50%",
          }
        }
      >
        <div
          style={
            Object {
              "height": "100%",
              "left": 0,
              "margin": 0,
              "padding": 0,
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </main>
</div>
`;
